### PR TITLE
fix: type errors and husky

### PR DESCRIPTION
**Pull request by pvinis**
### Description

This PR resolves []

we had a bunch of type errors. I fixed them and added a prepush hook to check, so we keep it that way.

### PR Checklist

- [ ] I tested my changes on **iOS** and **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[docs]: /docs/README.md

### CODE REVIEW

File: src/app/screens/HomeTabs/Artists/ArtistTabs/ArtistArtworks/EmailScreen/EmailScreen.tsx [hunk 1]
```diff
@@ -64,7 +64,7 @@ export const EmailScreen = () => {
             navigateTo={() => navigation.navigate("MultipleArtworksAndArtists")}
           />
         </Flex>
-        <Spacer m={6} />
+        <Spacer y={6} />
```
Comment (rajsam003) [inline src/app/screens/HomeTabs/Artists/ArtistTabs/ArtistArtworks/EmailScreen/EmailScreen.tsx:67]: 👀
While working on this PR https://github.com/artsy/energy/pull/273/files
I noticed that we cannot replace `m` with `y`. `y` gets a higher value than `m`.

File: src/app/screens/HomeTabs/Artists/ArtistTabs/ArtistTabs.tsx [hunk 2]
```diff
@@ -15,13 +17,11 @@ import { BriefcaseIcon, Screen, Button, EnvelopeIcon, Flex, Text } from "palette
...
 import { ArtistDocuments } from "./ArtistDocuments/ArtistDocuments"
 import { ArtistShows } from "./ArtistShows/ArtistShows"
-import { graphql, useLazyLoadQuery } from "react-relay"
-import { ArtistTabsQuery } from "__generated__/ArtistTabsQuery.graphql"
 
 type ArtistTabsRoute = RouteProp<HomeTabsScreens, "ArtistTabs">
 
 export const ArtistTabs = () => {
-  const { slug } = useRoute<ArtistTabsRoute>().params
+  const { slug, name } = useRoute<ArtistTabsRoute>().params
```
Comment (rajsam003) [inline src/app/screens/HomeTabs/Artists/ArtistTabs/ArtistTabs.tsx:24]: thanks 🙏🏽

File: src/app/screens/HomeTabs/Albums/AlbumArtworks.tsx [hunk 3]
```diff
@@ -68,7 +68,7 @@ export const AlbumArtworks = () => {
           marginTop: space(2),
         }}
         numColumns={2}
-        data={album.artworkIds}
+        data={album.artworkIds!}
```
Comment (rajsam003) [inline src/app/screens/HomeTabs/Albums/AlbumArtworks.tsx:71]: I thought we should avoid doing this 🤔

**Comment by rajsam003** [comment]
Thanks for this PR Pavlos. 
I just added a few comments. Could you please address them?

File: src/app/screens/HomeTabs/Albums/AlbumArtworks.tsx (hunk 3 above)
Comment (pvinis) [reply src/app/screens/HomeTabs/Albums/AlbumArtworks.tsx:71]: we should, but this PR is about adding that type check and this is a quick and dirty fix

File: src/app/screens/HomeTabs/Artists/ArtistTabs/ArtistArtworks/EmailScreen/EmailScreen.tsx (hunk 1 above)
Comment (pvinis) [reply src/app/screens/HomeTabs/Artists/ArtistTabs/ArtistArtworks/EmailScreen/EmailScreen.tsx:67]: yep. I broke spacer. both m and y are wrong. I am preparing a PR to fix this.

**Pull request merged**